server/aesdsocket.c

```diff
@@ -0,0 +1,293 @@
+      syslog(LOG_ERR , "malloc\n");
+      return -1;
+  }
+
+  uint8_t packet_check_flag = 0;
+  int byte_count = 0;
+  int datapacket = 0;
+  int final_size = 1;
+  int realloc_int = 0;
+  static int file_size = 0;
+  while(1){
+      //Check for accept connection
+      accept_fd = accept(socket_fd , &client_addr, &addr_size);
+      if (accept_fd == -1){
+          syslog(LOG_ERR , "accept\n");
+          return -1;
+      }
+
+      // Print IP address
+      // The inet_toa function requires a sockaddr_in structure
+      // Reference: https://stackoverflow.com/questions/3060950/how-to-get-ip-address-from-sock-structure-in-c
+      struct sockaddr_in * ip_client = (struct sockaddr_in *)&client_addr;
+      syslog(LOG_INFO , "Accepted connection from %s\n", inet_ntoa(ip_client->sin_addr));
+
+      while(!packet_check_flag){
+          byte_count = recv(accept_fd , recv_buffer , BUFFER_SIZE , 0);
+          // printf("Byte count:%d\n", byte_count);
+          if ( byte_count == -1){
+              syslog(LOG_ERR , "recv\n");
+              return -1;
+          }
+          else{
+              final_size++;
+              for(datapacket = 0; datapacket < BUFFER_SIZE; datapacket++){      //Check for newline character
+                  if(recv_buffer[datapacket] == '\n'){
+                      packet_check_flag = 1;
+                      break;
+                  }
+              }
+              memcpy((receive_buffer + (BUFFER_SIZE * realloc_int)) , recv_buffer , BUFFER_SIZE);
+              if(final_size >=BUFFER_SIZE) //Check for buffer is able to reallocate or not
+                {
+                  if(!(chec_ptr = realloc(receive_buffer, BUFFER_SIZE + final_size + 1)))
+                    {
+                      syslog(LOG_ERR , "realloc\n");
+                      return -1;
+                    }
+                  else
+                    {
+                      receive_buffer = chec_ptr;
+                      strncat(receive_buffer, recv_buffer, final_size);
+                    }
+
+                }
+              if(!packet_check_flag)    //Check for packet check flag status
+                {
+                  receive_buffer = (char *)realloc(receive_buffer , (sizeof(char) * (BUFFER_SIZE * final_size)));
+
+                  if (receive_buffer == NULL){
+
+                      syslog(LOG_ERR , "realloc\n");
+                  }
+                  else{
+                      realloc_int++;
+                  }
+                }
+
+          }
+      }
+
+      packet_check_flag = 0;
+      //Write the buffer data to the file
+      int buffer_len = ((datapacket + 1) + (BUFFER_SIZE * realloc_int));
+      int write_len = write(open_fd , receive_buffer , buffer_len);
+      if(write_len == -1){
+          syslog(LOG_ERR , "write\n");
+          return -1;
+      }
+      file_size += write_len;
+      realloc_int = 0, final_size = 1;
+      char *send_buffer = (char *)malloc(sizeof(char) * file_size);
```

Comment:
@Rajs060699 what assumption are you making here about the size of the process heap relative to the size of all packets sent?